Preliminary module for LvB MSc Theses LDA

anasashb opened the pull request:
By Bachana Anasashvili, Timur Tenishev and Yaman Tanilkan for the DEDA class

Progress so far:
Includes thesis scraper
includes language detector
Includes text preprocessing function
Includes custom designed CorpusMaker class
Includes custom designed user-friendly interface for doing LDA and LDA with gridsearch (LDA class)
Includes demo notebook with trial runs on the LDA
Includes pickled corpus, plots, some more outputs.
-----
To do:
requirements and pipfile will be updated
Demo Notebook and outputs will be updated after we run the grid search for LDA
A class will be set up for DTM and included in the demo notebook.

anasashb committed:
Create DEDA_class_SoSe2023_LDA_MSc_Theses

anasashb committed:
Delete DEDA_class_SoSe2023_LDA_MSc_Theses

anasashb committed:
Added initial module from local path

anasashb committed:
Addedd pickled texts

Needed for LDA grid search later on

anasashb committed:
Turned corpus maker into a class

anasashb committed:
minor update to text_prep

anasashb committed:
Custom LDA class with grid search, plotting, viz

anasashb committed:
More concise demo notebook

anasashb committed:
old demo notebook renamed to progress notebook

for transparency and keeping track of work

anasashb committed:
Full walkthrough in Demo Notebook

anasashb committed:
Added .simple_fit() method

anasashb committed:
Output folder now automatic

anasashb committed:
output folder now automatic

anasashb committed:
LDA topics added

Includes LDA topics from two trial runs: one for .simple_fit() and one for .build_best_model(). Also includes scores from the demo grid search.

anasashb committed:
New plots

Includes plots generated by CorpusMaker and LDA classes, including .html visualizations of the two LDA models fitted.

anasashb committed:
Output path to save LDA models

Includes saved LDA models from trial run of .simple_fit(0 and .build_best_model()

anasashb committed:
Dir for pickled data

ttenishev committed:
Updated corpus maker

Corpusmaker class now accomodates date assignments

ttenishev committed:
LDA seq model introduced

Updated file formating and included dates of thesis submission. Ldaseqmodel for DTM introduced

ttenishev committed:
Updated pkl files

Dates included, corpus and texts are stored chronologically now

ttenishev committed:
Gensim results

Updated after reorganizing corpus

ttenishev committed:
New function outputs

Automatic outputs from all functions after reorganizing corpus and introducing dates

ttenishev commented:
**Done:**

- Demo Notebook and outputs updated after the grid search for LDA had been run
- Included LDA seq model in the demo notebook

**To do:**

- Optimize DTM computing time and visualize the results of time topic distribution

ttenishev committed:
Topic evolution graphs included

Updates 1:
1. Random state given
2. Ldaseqmodel uses parameters from the best model of the grid search of non-dynamic LDA
3. Topic evolutions graphs are shown and the corresponding csv files are saved

anasashb committed:
Deleted some things

anasashb committed:
Deleted some more things

anasashb committed:
Deleted even more things

anasashb committed:
Deleted even more other things
